sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -714,8 +716,6 @@ private void SetNameFieldsIfNull()
             bool async = true,
             CancellationToken cancellationToken = default)
         {
-            BlobErrors.VerifyHttpsCustomerProvidedKey(Uri, CustomerProvidedKey);
```

Comment:
I moved this check to the client constructors instead of on individual APIs.

Reply:
What if someone calls the token/CPK ctor with https, but then modifies the Uri to use http. Are we concerned that they would now be able to call into the individual APIs without hitting an exception?